### PR TITLE
Jacob Yarborough DPL PR

### PR DESCRIPTION
**Pull request by jacobyarborough**

### CODE REVIEW

File: lib/library.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @authors.each do |author|
+      author.books.each do |book|
+        @book_array.push(book)
+      end
+    end
+    @book_array
+  end
+
+  def publication_time_frame_for(author)
+    return_hash = {}
```
Comment (dcoleman21) [inline lib/library.rb:26]: So this method is currently returning 
```
{:end=>"", :start=>"9999999999999"}
```
Here is another possible solution:
```
sorted_books = author.books.sort_by do |book|
     book.publication_year
end
{
     start: sorted_books.first.publication_year,
     end: sorted_books.last.publication_year
 }
```

File: spec/library_spec.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
+require 'rspec'
```
Comment (dcoleman21) [inline spec/library_spec.rb:1]: Nice job writing these tests.

**Review by dcoleman21 — COMMENTED** [review]
Overall nice job. I left you comments to review. We will be going over this tomorrow. 
Some parts were a little tricky the way it was written. 
Score: Pass

File: lib/library.rb (hunk 1 above)
Comment (jacobyarborough) [reply lib/library.rb:26]: Dani, 

Thanks for all the great feedback! I am wondering for this specifically why you were getting {:end=>"", :start=>"9999999999999"}? My. test for this method is expecting the following: 
`expect(dpl.publication_time_frame_for(charlotte_bronte)).to eq({:start=>"1847", :end=>"1857"})
      expect(dpl.publication_time_frame_for(harper_lee)).to eq({:start=>"1960", :end=>"1960"})` 
 This is passing in my rspec. Did I write these tests incorrectly? Like is it not actually retuning what I am expecting, but still somehow passing?

Comment (dcoleman21) [reply lib/library.rb:26]: Hey Jacob, I looked into this and there was an error in the spec_harness. What you have here does work. I appreciate your patience in this. Thank you for reaching out to clarify.

**Pull request closed without merging**